lib/library/library.ts

```diff
@@ -0,0 +1,489 @@
+import { Namesake } from '../npc-generation/_common'
+import { ThresholdTable } from '../src/rollFromTable'
+
+import { random } from '../src/random'
+import { createTippyFull } from '../src/tippy'
+import { Languages } from '../inactive/languages'
+
+interface BookData {
+  title: string
+  /** The Dewey system hasn't been invented yet, but... */
+  category: BookCategories[]
+  /** The blurb, a short summary. */
+  summary: string
+  /** If the book has an excerpt attached, it goes in contents. */
+  contents?: string
+  language?: Languages
+  condition?: string[]
+  cover?: string[]
+  author?: Namesake
+}
+
+type BookCategories =
```

Comment:
If you select this chunk of code and run *format selection*, it will standardize the typescript formatting.